File: backend/application/services/IUserService.cs
using application.dtos;
using core.models;

namespace application.services;

public interface IUserService
{
    public UserDto AddUser(UserCreationDto user);
    public UserDto? GetUser(string id);
    public UserDto? GetUserByEmail(string email);
    public void DeleteUser(string id);
    public List<GroupDto>? GetGroupsForUser(string id);
    public bool ValidatePassword(string password, string email);
}

File: frontend/file_samurai_frontend/src/services/IUserService.ts
import {User} from "../models/user.model";

export interface IUserService {
    getUserByToken(): Promise<User>;

    getUserIfNullRegister(email: string): Promise<User>;

    registerUser(email: string, hashedPassword: string, salt: string): Promise<User>;

    validatePassword(password: string): Promise<boolean>;
}